test_search.c

```diff
@@ -0,0 +1,15 @@
+#ifndef __INCLUDES_H_
+#define __INCLUDES_H_
+    #include "naive_search.h"
+    #include "search_utils.h"
+#endif
+
+int main(int argc, char* argv[]) {
+
+    // todo: import and use DNA data for testing.
+    char* pattern = "the";
+    char* target = "the dog chased the cat but the cat was too fast.";
+
+    SearchResult naiveSerialSearchResult = naiveSearch(pattern, 3, target, 48);
+    printSearchResults(naiveSerialSearchResult);
+}
```

Comment:
Once the other algorithms are implemented, we can add 2 lines like line 13/14 here for each algorithm to execute the search and print the results.